sklearn/decomposition/_dict_learning.py

```diff
@@ -1844,39 +2226,108 @@ def fit(self, X, y=None):
+        dict_buffer = dictionary.copy()
+
+        if self.shuffle:
+            X_train = X.copy()
+            self._random_state.shuffle(X_train)
+        else:
+            X_train = X
+
+        if self.verbose:
+            print("[dict_learning]")
+
+        # Inner stats
+        self._inner_stats = (
+            np.zeros((self._n_components, self._n_components)),
+            np.zeros((n_features, self._n_components)),
         )
-        self.components_ = U
-        # Keep track of the state of the algorithm to be able to do
-        # some online fitting (partial_fit)
-        self.inner_stats_ = (A, B)
-        self.iter_offset_ = self.n_iter
-        self.random_state_ = random_state
+
+        if self.max_iter is not None:
+
+            # Attributes to monitor the convergence
+            self._ewa_cost = None
+            self._ewa_cost_min = None
+            self._no_improvement = 0
+
+            batches = gen_batches(n_samples, self._batch_size)
+            batches = itertools.cycle(batches)
+            n_steps_per_epoch = int(np.ceil(n_samples / self._batch_size))
+            n_steps = self.max_iter * n_steps_per_epoch
+
+            # allow max_iter = 0
+            i = -1
+            import time
+
+            t = 0
+            for i, batch in zip(range(n_steps), batches):
+                t1 = time.time()
+                this_X = X_train[batch]
+
+                batch_cost = self._minibatch_step(
+                    this_X, dictionary, self._random_state, i
+                )
+                t += time.time() - t1
+                if self._minibatch_convergence(
+                    this_X, batch_cost, dictionary, dict_buffer, n_samples, i, n_steps
+                ):
+                    break
+                print(f"[{self._ewa_cost},{t}],")
+
+                if self.callback is not None:
+                    self.callback(locals())
+
+                dict_buffer[:] = dictionary
+
+            self.n_steps_ = i + 1
+            self.n_iter_ = np.ceil((i + 1) / n_steps_per_epoch)
+        else:
+            # TODO remove this branch in 1.3
+            if self.n_iter != "deprecated":
+                warnings.warn(
+                    "'n_iter' is deprecated in version 1.1 and will be removed"
+                    " in version 1.3. Use 'max_iter' instead.",
+                    FutureWarning,
+                )
+                n_iter = self.n_iter
+            else:
+                n_iter = 1000
+
+            batches = gen_batches(n_samples, self._batch_size)
+            batches = itertools.cycle(batches)
+
+            for i, batch in zip(range(n_iter), batches):
+                self._minibatch_step(X_train[batch], dictionary, self._random_state, i)
+
+                trigger_verbose = self.verbose and i % ceil(100.0 / self.verbose) == 0
+                if self.verbose > 10 or trigger_verbose:
+                    print(f"{i} batches processed.")
```

Comment:
Please extend `test_dict_learning_online_verbosity` to cover this case.

Reply:
This is in the old_behavior. I think it's ok to not try to cover this now since it will be removed in 2 versions anyway